### ISSUE TITLE
Sandbox Tools

### ISSUE THREAD

**Issue by Ashotep** [peterhaneve/ONIMods#290]
CTD when using the reveal tool.

## Expected Behavior
Reveal blacked out sections of the map
Spaced out DLC installed

## Current Behavior
CTD

## Mod Versions
Steam 3.6
yes

## Steps to Reproduce
1.  Start a new map
2.  Open Sandbox mode
3.  Select reveal tool
4.  Reveal map till CTD

## Output Log
Crash Report:

```
NotSupportedException: Collection is read-only

(wrapper managed-to-managed) Tag[].System.Collections.Generic.ICollection`1.Clear()
PeterHan.SandboxTools.SandboxToolsPatches+BuildingDef_Build_Patch.Prefix (BuildingDef __instance, System.Collections.Generic.IList`1[T] selected_elements, System.Single& temperature) (at <f0b0964f6d464c048b8a57678de5cac4>:0)
(wrapper dynamic-method) BuildingDef.BuildingDef.Build_Patch1(BuildingDef,int,Orientation,Storage,System.Collections.Generic.IList`1<Tag>,single,bool,single)
Scenario.PlaceBuilding (System.Int32 root_cell, System.Int32 x, System.Int32 y, System.String prefab_id, SimHashes element) (at <a0b1e63c8d674793957c96bc63afc3da>:0)
TemplateLoader.PlaceBuilding (TemplateClasses.Prefab prefab, System.Int32 root_cell) (at <a0b1e63c8d674793957c96bc63afc3da>:0)
WorldGenSpawner+Spawnable.Spawn () (at <a0b1e63c8d674793957c96bc63afc3da>:0)
WorldGenSpawner+Spawnable.TrySpawn () (at <a0b1e63c8d674793957c96bc63afc3da>:0)
WorldGenSpawner+Spawnable.OnReveal (System.Object data) (at <a0b1e63c8d674793957c96bc63afc3da>:0)
ScenePartitioner.RunEntries (System.Collections.Generic.List`1[T] gathered_entries, System.Object event_data) (at <a0b1e63c8d674793957c96bc63afc3da>:0)
ScenePartitioner.TriggerEvent (System.Collections.Generic.List`1[T] cells, ScenePartitionerLayer layer, System.Object event_data) (at <a0b1e63c8d674793957c96bc63afc3da>:0)
GameScenePartitioner.TriggerEvent (System.Collections.Generic.List`1[T] cells, ScenePartitionerLayer layer, System.Object event_data) (at <a0b1e63c8d674793957c96bc63afc3da>:0)
World.LateUpdate () (at <a0b1e63c8d674793957c96bc63afc3da>:0)

Build: U43-531669-SD
```

Log attatched
[Player.log](https://github.com/peterhaneve/ONIMods/files/10200581/Player.log)

**Comment by peterhaneve** [comment]
This crash has been fixed in version 3.7.0.0. It is likely that you will need to use Mod Updater to receive this version.

**Comment by Ashotep** [comment]
Thanks
